html/pfappserver/lib/pfappserver/Model/Search/User.pm

```diff
@@ -14,162 +15,240 @@ User
-    $count = $count->{count};
-    $results{count} = $count;
-    $results{page_count} = calc_page_count($count, $per_page);
-    $results{per_page} = $per_page;
-    $results{page_num} = $page_num;
-    return \%results;
+=cut
+
+sub _build_limit {
+    my ($self, $c, $params, $search_info) = @_;
+    my $page_num = $params->{page_num} || 1;
+    my $limit  = $params->{per_page} || 25;
+    my $offset = (( $page_num - 1 ) * $limit);
+    $search_info->{-limit} = $limit;
+    $search_info->{-offset} = $offset;
 }
 
-sub add_searches {
-    my ($self,$builder,$params) = @_;
-    my @searches = map {$self->process_query($_)} @{$params->{searches}};
-    my $all_or_any = $params->{all_or_any};
-    if ($all_or_any eq 'any' ) {
-        $all_or_any = 'or';
-    } else {
-        $all_or_any = 'and';
-    }
-    if(@searches) {
-        $builder->where('(');
-        $builder->where($all_or_any)->where(@$_) for @searches;
-        $builder->where(')');
+
+=head2 _build_order_by
+
+Build the order by clause of the query
+
+=cut
+
+sub _build_order_by {
+    my ($self, $c, $params, $search_info) = @_;
+    my ($by, $direction) = @$params{qw(by direction)};
+    $by //= 'person.pid';
+    $direction //= 'ASC';
+    $direction = uc($direction);
+    if ($direction ne 'ASC' && $direction ne 'DESC') {
+        $direction = 'ASC';
     }
+    $search_info->{-order_by} = ["$by $direction"];
+}
+
+our %OP_MAP = (
+    equal       => '=',
+    not_equal   => '<>',
+    not_like    => 'NOT LIKE',
+    like        => 'LIKE',
+    ends_with   => 'LIKE',
+    starts_with => 'LIKE',
+    in          => 'IN',
+    not_in      => 'NOT IN',
+);
+
+=head2 _build_clause
+
+Build clause from the query
+
+=cut
 
+sub _build_clause {
+    my ($self, $query) = @_;
+    my $op = $query->{op};
+    my $value = $query->{value};
+    my $name = $query->{name};
+    return unless defined $op && defined $name;
+    return unless defined $value;
+    $name = $self->fixup_name($name);
+    $value //= '';
+    die "$op is not a supported search operation"
+        unless exists $OP_MAP{$op};
+    my $sql_op = $OP_MAP{$op};
+    if($sql_op eq 'LIKE' || $sql_op eq 'NOT LIKE') {
+        #escaping the % and _ charcaters
+        my $escaped = $value =~ s/([%_])/\\$1/g;
+        if($op eq 'like' || $op eq 'not_like') {
+            $value = "\%$value\%";
+        } elsif ($op eq 'starts_with') {
+            $value = "$value\%";
+        } elsif ($op eq 'ends_with') {
+            $value = "\%$value";
+        }
+        if ($escaped) {
+            return { $name => { like => \[q{? ESCAPE '\'}, $value] } };
+        }
+    }
+    return {$name => {$sql_op => $value}};
 }
 
-sub add_order_by {
-    my ($self, $builder, $params) = @_;
-    my ($by, $direction) = @$params{qw(by direction)};
-    if ($by && $direction) {
-        $by = $COLUMN_MAP{$by} if (exists $COLUMN_MAP{$by});
-        $builder->order_by($by, $direction);
+{
+
+my %COLUMN_MAP = (
```

Comment:
This should either be at the top of the file or inside the sub, but declaring a constant in the middle of the file seems a bit odd to me